sandbox/concurrent_bert_etl/app_all_lambdas/jobs.py

```diff
@@ -0,0 +1,542 @@
+    outbucket = s3.Bucket(name=os.environ.get('CACHEBUCKETNAME'))
+    homedir = os.environ.get('HOME')
+
+    work_queue, done_queue, ologger = utils.comm_binders(
+        bert_tess_fullframe_main_2)
+
+    # Example event:
+    # {
+    #   "tic_id": "25155310",
+    #   "sec_id": "tess-s0001-4-1",
+    #   "ra": 63.3739396231274,
+    #   "dec": -69.226822697583,
+    #   "radius": 2.5,
+    #   "cutout_width": 30,
+    #   "use_cache": "true"
+    # }
+    #
+    # work_queue populated by calling Lambda
+    for event in work_queue:
+        tic_id = event['tic_id']
+        sec_id = event['sec_id']
+
+        basename = f'{sec_id}_s3_uris.txt'  # noqa
+        filename = os.path.join(homedir, basename)
+
+        try:
+            # Check if URI list already cached.
+            # According to MAST, there is no need to invalidate cache here.
+            ologger.info(f'Attempting to download {basename} from S3')
+            outbucket.download_file(
+                basename, filename,
+                ExtraArgs={"RequestPayer": "requester"})
+        except Exception:
+            # Find full frame dataset for the observation ID.
+            ologger.info('Started quering Observations...')
+            obs_table = Observations.query_criteria(obs_id=sec_id)
+            products = Observations.get_product_list(obs_table)
+            filtered = Observations.filter_products(
+                products, productSubGroupDescription="FFIC",
+                mrp_only=False)
+
+            # Use AWS S3 bucket to pull data from.
+            Observations.enable_cloud_dataset(verbose=False)
+            ologger.info('Started obtaining cloud URIs...')
+            t_start = time.time()
+            s3_urls = Observations.get_cloud_uris(
+                filtered, include_bucket=False)
+            t_end = time.time()
+            ologger.info(f'Got {len(s3_urls)} URIs in {t_end - t_start} s')
+
+            # Upload URI list to cache.
+            with open(filename, 'w') as fout:
+                for url in s3_urls:
+                    fout.write(url + os.linesep)
+            try:
+                outbucket.upload_file(
+                    filename, basename,
+                    ExtraArgs={"RequestPayer": "requester"})
+            except Exception as exc:
+                ologger.error(str(exc))
+            else:
+                ologger.info(f'Uploaded {basename} to S3')
+        else:
+            # Use cache if it exists.
+            with open(filename, 'r') as fin:
+                s3_urls = [url.strip() for url in fin.readlines()]
+            ologger.info(f'Read {len(s3_urls)} URIs from {basename}')
+        finally:
+            # Clean up
+            if os.path.exists(filename):
+                os.remove(filename)
+
+        ra = float(event['ra'])
+        dec = float(event['dec'])
+
+        # TODO: Cache WCS for a given sector/camera/ccd combo.
+        # TODO: How to make sure WCS is good? 10% of FFIs might have bad WCS.
```

Comment:
NOTE: If `WCSAXES=2`, the WCS is good (source: MIT and C. Brasseur). Can use `astropy.wcs.WCS` to do FITS WCS I/O.